amaranth/sim/_pyrtl.py

```diff
@@ -224,20 +225,33 @@ def on_Repl(self, value):
         return f"0"
 
     def on_ArrayProxy(self, value):
+        use_pm = sys.version_info >= (3, 10)
```

Comment:
Can you make this a private module-level constant? Something like `_USE_PATTERN_MATCHING = (sys.version_info >= (3, 10))` after the `__all__` definition.